state-chain/pallets/cf-elections/src/electoral_systems/solana_swap_accounts_tracking.rs

```diff
@@ -0,0 +1,274 @@
+	for SolanaVaultSwapAccounts<Account, SwapDetails, BlockNumber, Settings, Hook, ValidatorId, E>
+{
+	type ValidatorId = ValidatorId;
+	type ElectoralUnsynchronisedState = SolanaVaultSwapAccountsLastClosedAt<BlockNumber>;
+	type ElectoralUnsynchronisedStateMapKey = ();
+	type ElectoralUnsynchronisedStateMapValue = ();
+
+	type ElectoralUnsynchronisedSettings = ();
+	type ElectoralSettings = Settings;
+	type ElectionIdentifierExtra = ();
+	type ElectionProperties = SolanaVaultSwapsKnownAccounts<Account>;
+	type ElectionState = ();
+	type Vote = vote_storage::bitmap::Bitmap<SolanaVaultSwapsVote<Account, SwapDetails>>;
+	type Consensus = SolanaVaultSwapsVote<Account, SwapDetails>;
+	type OnFinalizeContext = BlockNumber;
+	type OnFinalizeReturn = ();
+
+	fn generate_vote_properties(
+		_election_identifier: ElectionIdentifier<Self::ElectionIdentifierExtra>,
+		_previous_vote: Option<(VotePropertiesOf<Self>, AuthorityVoteOf<Self>)>,
+		_vote: &<Self::Vote as VoteStorage>::PartialVote,
+	) -> Result<VotePropertiesOf<Self>, CorruptStorageError> {
+		Ok(())
+	}
+
+	fn is_vote_desired<ElectionAccess: ElectionReadAccess<ElectoralSystem = Self>>(
+		_election_identifier_with_extra: crate::electoral_system::ElectionIdentifierOf<Self>,
+		_election_access: &ElectionAccess,
+		_current_vote: Option<(VotePropertiesOf<Self>, AuthorityVoteOf<Self>)>,
+	) -> Result<bool, CorruptStorageError> {
+		Ok(true)
+	}
+
+	fn is_vote_needed(
+		(_, current_partial_vote, _): (
+			VotePropertiesOf<Self>,
+			<Self::Vote as VoteStorage>::PartialVote,
+			AuthorityVoteOf<Self>,
+		),
+		(proposed_partial_vote, _): (
+			<Self::Vote as VoteStorage>::PartialVote,
+			<Self::Vote as VoteStorage>::Vote,
+		),
+	) -> bool {
+		current_partial_vote != proposed_partial_vote
+	}
+
+	fn on_finalize<ElectoralAccess: ElectoralWriteAccess<ElectoralSystem = Self>>(
+		electoral_access: &mut ElectoralAccess,
+		election_identifiers: Vec<ElectionIdentifier<Self::ElectionIdentifierExtra>>,
+		current_block_number: &Self::OnFinalizeContext,
+	) -> Result<Self::OnFinalizeReturn, CorruptStorageError> {
+		if let Some(election_identifier) = election_identifiers
+			.into_iter()
+			.at_most_one()
+			.map_err(|_| CorruptStorageError::new())?
+		{
+			let mut election_access = electoral_access.election_mut(election_identifier)?;
+			if let Some(consensus) = election_access.check_consensus()?.has_consensus() {
+				let mut known_accounts = election_access.properties()?;
+				election_access.delete();
+				known_accounts.witnessed_open_accounts.extend(consensus.new_accounts.iter().map(
+					|(account, swap_details)| {
+						Hook::initiate_vault_swap((*swap_details).clone());
+						(*account).clone()
+					},
+				));
+				consensus.confirm_closed_accounts.into_iter().for_each(|acc| {
+					known_accounts.closure_initiated_accounts.remove(&acc);
+				});
+
+				if Hook::get_number_of_available_sol_nonce_accounts() >
+					NONCE_AVAILABILITY_THRESHOLD_FOR_INITIATING_SWAP_ACCOUNT_CLOSURES &&
```

Comment:
comment here would be useful, just so it's clear that it's just to provide better service to users, there's not some fundamental reason